main.go

```diff
@@ -280,49 +278,41 @@ func (s *Session) firstNav(ctx context.Context) error {
 	return nil
 }
 
-// navToEnd waits for the page to be ready to receive scroll key events, by
-// trying to select an item with the right arrow key, and then scrolls down to the
-// end of the page, i.e. to the oldest items.
+// navToEnd selects the last item in the page
+//  by repeatedly advances the selected item with
+//  - kb.ArrowRight (which causes an initial selection, and/or advances it by one)
+//  - kb.End which scrolls to the end of the page, and advances the selected item.
+// Note timing is important, because when the kb.End causes significant scrolling,
+//  the active element become undefined for a certain time, in that case, we get an error (ignore), sleep, and retry.
+// The termnation criteria is that the selected item (document.activeElement.href) is stable for >2 iterations
 func navToEnd(ctx context.Context) error {
-	// wait for page to be loaded, i.e. that we can make an element active by using
-	// the right arrow key.
+	var prev, active string
+	lastRepeated := 0
 	for {
 		chromedp.KeyEvent(kb.ArrowRight).Do(ctx)
+		chromedp.KeyEvent(kb.End).Do(ctx)
 		time.Sleep(tick)
-		var ids []cdp.NodeID
-		if err := chromedp.Run(ctx,
-			chromedp.NodeIDs(`document.activeElement`, &ids, chromedp.ByJSPath)); err != nil {
-			return err
+
+		if err := chromedp.Evaluate(`document.activeElement.href`, &active).Do(ctx); err != nil {
+			time.Sleep(tick) // this extra sleep is important: after the kb.End, it sometimes takes a while for the active element to be reset
+			continue         // ignore this error: no active element, or active element has no href
 		}
-		if len(ids) > 0 {
-			if *verboseFlag {
-				log.Printf("We are ready, because element %v is selected", ids[0])
-			}
-			break
+		if active == prev {
+			lastRepeated++
+		} else {
+			lastRepeated = 0
 		}
-		time.Sleep(tick)
-	}
-
-	// try jumping to the end of the page. detect we are there and have stopped
-	// moving when two consecutive screenshots are identical.
-	var previousScr, scr []byte
-	for {
-		chromedp.KeyEvent(kb.PageDown).Do(ctx)
-		chromedp.KeyEvent(kb.End).Do(ctx)
-		chromedp.CaptureScreenshot(&scr).Do(ctx)
-		if previousScr == nil {
-			previousScr = scr
-			continue
+		if *verboseFlag {
+			log.Printf("** navToEnd:activeElt %s %d", active, lastRepeated)
 		}
-		if bytes.Equal(previousScr, scr) {
+		if lastRepeated > 2 {
 			break
 		}
-		previousScr = scr
-		time.Sleep(tick)
+		prev = active
+		// time.Sleep(tick)
 	}
-
 	if *verboseFlag {
-		log.Printf("Successfully jumped to the end")
+		log.Printf("Successfully jumped to the end: %s", active)
```

Comment:
Same comment as last time, if active is not actually the very last element.

Reply:
Now that the we are using `document.activeElement`, and `lastRepeated` in our termination criteria, we are actually sure that we do have the last element.

Reply:
well, then my other remark from the previous patch stands again I think ;P
i.e.: are we now not redundant with some of what navToLast does?

Reply:
You are right that It is redundant with some of what `navToLast` does, but certainly compatible with it.

I thought we were trying to keep the change minimal (the PR is trying to replace the screenshot stuff).

Do you want to also modify `navToLast` to only what is essential to work with this new version of `navToEnd`?

Reply:
> You are right that It is redundant with some of what `navToLast` does, but certainly compatible with it.
> 
> I thought we were trying to keep the change minimal (the PR is trying to replace the screenshot stuff).

Yes, ideally we always want to keep a PR about one change and one change only, and not about several unrelated changes. However, if your fix to remove the screenshots makes navToLast obsolete and useless (which I think it does), then it's actually a good thing to remove it as part of the same PR.

> Do you want to also modify `navToLast` to only what is essential to work with this new version of `navToEnd`?

Yep, or even better, let's fully remove it if we can. But if you're tired of this review, I can do it in a subsequent PR, no worries.